examples/mnist/examples/mnist.rs

```diff
@@ -24,9 +24,15 @@ mod tch_gpu {
     pub fn run() {
         #[cfg(not(target_os = "macos"))]
         let device = TchDevice::Cuda(0);
+        #[cfg(not(target_os = "macos"))]
+        type FloatElement = burn::tensor::f16;
```

Comment:
We need to test f16 on GPU without f16 native support. My linux  GPU (GeForce GTX 1080) does not have f16 native.  Maybe it works without overflow.

Reply:
I'll put f32 by default then, it's not a big deal :)